src/services/balancer/subgraph/subgraph-fallback.service.ts

```diff
@@ -0,0 +1,57 @@
+import { captureException } from '@sentry/browser';
+import axios, { AxiosResponse } from 'axios';
+import { computed, ref } from 'vue';
+
+import { configService } from '@/services/config/config.service';
+
+class SubgraphFallbackService {
+  private subgraphUrlIndex = ref(0);
+
+  public isSingleSubgraphUrl = computed(() => this.subgraphUrls.length === 0);
+  public currentSubGraphUrl = computed(() => {
+    if (this.isSingleSubgraphUrl.value) {
+      return this.url;
+    }
+    return this.subgraphUrls[this.subgraphUrlIndex.value];
+  });
+
+  constructor(
+    private readonly url = configService.subgraph,
+    private readonly subgraphUrls = configService.subgraphUrls || []
+  ) {}
```

Comment:
Can we not simplify this to 
```suggestion
  constructor(
    private readonly urls = [configService.subgraph, ...configService.subgraphUrls]
  ) {}
```